src/cloud-resources/src/crd/materialize.rs

```diff
@@ -109,6 +135,20 @@ pub mod v1alpha1 {
         // will populate any defaults.
         #[serde(default = "Uuid::new_v4")]
         pub environment_id: Uuid,
+
+        // The configuration for generating an x509 certificate using cert-manager for balancerd
+        // to present to incoming connections.
+        // The dns_names and issuer_ref fields are required.
+        pub balancerd_external_certificate_spec: Option<MaterializeCertSpec>,
+        // The configuration for generating an x509 certificate using cert-manager for the console
+        // to present to incoming connections.
+        // The dns_names and issuer_ref fields are required.
+        // Not yet implemented.
+        pub console_external_certificate_spec: Option<MaterializeCertSpec>,
+        // The cert-manager Issuer or ClusterIssuer to use for database internal communication.
+        // The issuer_ref field is required.
+        // This currently is only used for environmentd, but will eventually support clusterd.
+        pub internal_certificate_spec: Option<MaterializeCertSpec>,
```

Comment:
thinking about it some more, i wonder how annoying it is going to be to have this solely configured at the environment level - it kind of feels like we want to be able to set defaults for all of these fields at the orchestratord level (as part of the helm chart config) because in the vast majority of cases, users are going to want all of the fields here (except possibly the dns names) to be enforced to be identical across environments.

Reply:
Do you think we should have one set of defaults passed to orchestratord from the helm chart, or do we need a set for each type of cert?